build.gradle

```diff
@@ -7,41 +12,175 @@ apply plugin: 'idea'
+        classpath "org.tmatesoft.svnkit:svnkit:${svnkitVersion}"
+    }
+}
+
+class SvnCheckout extends DefaultTask {
+
+    @Input
+    String svnRef = 'https://svn.apache.org/repos/asf/db/derby/code/branches/'
+
+    @Input
+    String branch = '10.16'
+
+    @OutputDirectory
+    File getCheckoutDir() {
+        return project.layout.buildDirectory.dir(branch).get().asFile
+    }
+
+    SvnCheckout() {
+        description = "Checkout a branch from svnRef"
+        group = "org.logstash.tooling"
+    }
+
+    @TaskAction
+    def checkout() {
+        SVNClientManager clientManager = SVNClientManager.newInstance();
+        SVNUpdateClient client = clientManager.getUpdateClient();
+        SVNURL svnurl = SVNURL.parseURIEncoded(svnRef + branch);
+
+        println "Starting checkout"
+        long revision = client.doCheckout(svnurl, checkoutDir, SVNRevision.HEAD,
+                SVNRevision.HEAD, SVNDepth.UNKNOWN, true);
+        println "Checked out at revision ${revision} in folder ${checkoutDir}"
+    }
+}
 
 repositories {
     mavenCentral()
 }
 
 
 dependencies {
-    implementation "org.apache.derby:derby:${DERBY_VERSION}"
-    implementation "org.apache.derby:derbyclient:${DERBY_VERSION}"
+//    implementation "org.apache.derby:derby:${DERBY_VERSION}"
+//    implementation "org.apache.derby:derbyclient:${DERBY_VERSION}"
+    implementation files('local_repository/derby-10.15.2.1.jar', 'local_repository/derbyclient-10.15.2.1.jar')
+}
+
+tasks.register("svnCheckout", SvnCheckout) {
+    branch = DERBY_MINOR
 }
 
-task generateGemJarRequiresFile {
+ant.lifecycleLogLevel = "INFO"
+
+tasks.register("importAnt") {
+    dependsOn svnCheckout
+
+    doLast {
+        ant.importBuild(layout.buildDirectory.file("${DERBY_MINOR}/build.xml").get().asFile) { antTargetName ->
+            'ant-' + antTargetName
+        }
+        println "Ant imported"
+    }
+}
+
+tasks.register("buildDerby") {
+    dependsOn importAnt
+
+    doLast {
+        ant.antProject.setBasedir(layout.buildDirectory.dir(DERBY_MINOR).get().asFile.toString())
+        ant.antProject.executeTarget("clobber")
+        ant.antProject.executeTarget("buildsource")
+        ant.antProject.executeTarget("buildjars")
+    }
+}
+
+def readFullDerbyVersion(String derbyVersionBranch) {
+    def releasePropertiesFile = layout.buildDirectory.file("${derbyVersionBranch}/tools/ant/properties/release.properties").get().asFile
+    def props = new Properties()
+    releasePropertiesFile.withInputStream { props.load(it) }
+
+    return props.getProperty('release.id.long')
+}
+
+tasks.register('deployLocallyDerbyArtifacts') {
+    dependsOn buildDerby
+    description = "Copy Derby and Derby Client in local repository, which can be used to resolve dependencies"
+
+    doLast {
+        String derbyFullVersion = readFullDerbyVersion(DERBY_MINOR)
+        copy {
+            from layout.buildDirectory.dir("${DERBY_MINOR}/jars/sane/derbyshared.jar")
+            into file('local_repository/')
```

Comment:
this folder is not being tracked so gradle clean does not delete it

Reply:
correct